File: learning-akka/src/test/java/book/java/ch1/AkkademyDbTest.java
package book.java.ch1;

import akka.actor.ActorRef;
import akka.actor.ActorSystem;
import akka.actor.Props;
import akka.testkit.TestActorRef;
import book.kotlin.messages.SetRequest;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class AkkademyDbTest {

	private ActorSystem system = ActorSystem.create();

	@Test
	public void itShouldPlaceKeyValueFromSetMessageIntoMap() {
		TestActorRef<AkkademyDb> actorRef = TestActorRef.create(system, Props.create(AkkademyDb.class));
		actorRef.tell(new SetRequest("key", "value"), ActorRef.noSender());
		AkkademyDb akkademyDb = actorRef.underlyingActor();
		assertEquals(akkademyDb.map.get("key"), "value");
	}

}

File: learning-akka/src/test/kotlin/book/kotlin/ch1/AkkademyDbTest.kt
package book.kotlin.ch1

import akka.actor.ActorSystem
import book.kotlin.messages.SetRequest
import book.kotlin.util.TestActorRef
import book.kotlin.util.tell
import org.jetbrains.spek.api.Spek
import org.jetbrains.spek.api.dsl.describe
import org.jetbrains.spek.api.dsl.it
import org.junit.Assert.assertEquals
import org.junit.platform.runner.JUnitPlatform
import org.junit.runner.RunWith

@RunWith(JUnitPlatform::class)
class AkkademyDbTest : Spek({

    val system = ActorSystem.create()

    describe("akkademyDb") {

        describe("given SetRequest") {

            it("should place key/value into map") {
                val actorRef = TestActorRef<AkkademyDb>(system)
                actorRef.tell(SetRequest("key", "value"))
                val akkademyDb = actorRef.underlyingActor()
                assertEquals(akkademyDb.map["key"], "value")
            }

        }

    }
})

